### PR TITLE
[FEATURE] Added Product OptionLabel Resolver for the ProductInterface…

### PR DESCRIPTION
**Pull request by lewisvoncken**
… and added the resolver to the manufacturer attribute in the graphql schema

### Fixed Issues (if relevant)

1. magento/magento2#<issue_number>: Issue title

### Manual testing scenarios (*)

1. use the following query and variables to retrieve the the product detail information
```
query productDetail($urlKey: String) {
  productDetail: products(filter: {url_key: {eq: $urlKey}}) {
    items {
      __typename
      sku
      name
      url_path
      manufacturer
    }
    __typename
  }
}
```
```
{"urlKey":"your-product-urlkey"}
```
2. You should see the manufacturer returned with the correct option label, for example:
```
{
  "data": {
    "productDetail": {
      "items": [
        {
          "__typename": "ConfigurableProduct",
          "sku": "123456",
          "name": "Your Product",
          "url_path": null,
          "manufacturer": "Mr. Lewis"
        }
      ],
      "__typename": "Products"
    }
  }
}
```

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/CatalogGraphQl/etc/schema.graphqls [hunk 1]
```diff
@@ -96,7 +96,7 @@ interface ProductInterface @typeResolver(class: "Magento\\CatalogGraphQl\\Model\
     tier_prices: [ProductTierPrices] @doc(description: "An array of ProductTierPrices objects.") @resolver(class: "Magento\\CatalogGraphQl\\Model\\Resolver\\Product\\TierPrices")
     price: ProductPrices @doc(description: "A ProductPrices object, indicating the price of an item.") @resolver(class: "Magento\\CatalogGraphQl\\Model\\Resolver\\Product\\Price")
     gift_message_available: String @doc(description: "Indicates whether a gift message is available.")
-    manufacturer: Int @doc(description: "A number representing the product's manufacturer.")
+    manufacturer: String @doc(description: "The label of the product's manufacturer.") @resolver(class: "Magento\\CatalogGraphQl\\Model\\Resolver\\Product\\OptionLabel")
```
Comment (lenaorobei) [inline app/code/Magento/CatalogGraphQl/etc/schema.graphqls:99]: It is a braking change.

**Comment by lenaorobei** [comment]
@lewisvoncken could you please provide some use cases since there was no original issue for this implementation.

**Comment by TomashKhamlai** [comment]
Please, address this issue
image

**Comment by lenaorobei** [comment]
@lewisvoncken closing this PR due to inactivity.
I you would like to continue working on it, please prepare use cases an and feel free to reopen it. Looks like this functionality can be implemented in the custom module.

Thanks.

**Pull request closed without merging**